source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -46,7 +46,9 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
       ssl_curves_(stat_name_set_->add("ssl.curves")),
       ssl_sigalgs_(stat_name_set_->add("ssl.sigalgs")) {
   const auto tls_certificates = config.tlsCertificates();
-  tls_contexts_.resize(std::max(static_cast<size_t>(1), tls_certificates.size()));
+// Use a single context for certificates instead of one context per certificate as
+// openssl handles the multi cert case automatically in the handshake
+  tls_contexts_.resize(1);
```

Comment:
if the context is always of size 1, why not drop the vector altogether?

Reply:
Yes I agree.

I had a look at the context-impl.cc and there are a few lines I want to be sure of before taking that route.

Reply:
There are a number of places in the context-impl.cc where tls_contexts_ are referenced in for loops, it should be low risk to dispose of the vector.

Reply:
Agreed with Dmitri we will drop the vector in the server context.